languages/typescript/tree-sitter/Boilerplate_tree_sitter_typescript.ml

```diff
@@ -0,0 +1,3187 @@
+(*
+   Coupling: If you change this header, you should change the
+   header for CST_tree_sitter_typescript.ml as well, and vice versa.
+
+   Treated similarly to `CST_tree_sitter_typescript.ml`.
+
+   Originally generated by ocaml-tree-sitter, this is the merger
+   of the two variants of the same file Boilerplate.ml:
+
+   - typescript/lib/Boilerplate.ml
+   - tsx/lib/Boilerplate.ml
+
+   Look at the original javascript grammar in 'define-grammar.js' to identify
+   and reconciliate the differences. There's one spot with a conditional,
+   which results in different variants for the expression type.
+
+   How to update:
+
+   1. Make a copy of the original for reference:
+
+        cp Boilerplate_tree_sitter_typescript.ml Boilerplate_tree_sitter_typescript.ml.old
+
+   2. Replace this file with the Boilerplate.ml that was generated for tsx
+      (because it has more definitions than the one for pure typescript):
+
+        cp ../../tree-sitter-lang/semgrep-tsx/lib/Boilerplate.ml \
+          Boilerplate_tree_sitter_typescript.ml
+
+   3. Edit the new file Boilerplate_tree_sitter_typescript.ml:
+      a. Copy this header.
+      b. Add the "type assertion" case. Look for 'Type_asse' in
+         the new generated file semgrep-typescript/lib/Boilerplate.ml
+*)
+(**
+   Boilerplate to be used as a template when mapping the tsx + typescript CST
+   to another type of tree.
+*)
```

Comment:
I would add something like this:

> We need this hand-modified file because it provides the `dump_tree` function that works on the merged TSX/TypeScript CST type.